v2/cf-nodejs-logging-support/src/lib/logger/record-factory.ts

```diff
@@ -60,12 +87,32 @@ export default class RecordFactory {
                 }
             }
 
-            // TO DO: sources as array case
+            // if envVarRedacted and not enabled
+            if (field._meta?.isRedacted == true && field._meta?.isEnabled == false) {
+                // if field value is present set to "redacted", if not ommit
+                if (record[field.name]) {
+                    record[field.name] = this.REDUCED_PLACEHOLDER;
+                } else {
+                    record[field.name] = undefined;
+                }
+                return;
+            }
```

Comment:
If `isEnabled` is `false`, we should not log the field even if `isRedacted` is `true`. See comment above on simplifying logic.